packages/@lwc/template-compiler/src/parser/expression.ts

```diff
@@ -10,19 +10,20 @@ import { ParserDiagnostics, invariant } from '@lwc/errors';
 import * as t from '../shared/estree';
 import { NormalizedConfig } from '../config';
 import { Expression, Identifier, SourceLocation } from '../shared/types';
+import { validateExpressionAst } from './expression-complex';
+import { TMPL_EXPR_ECMASCRIPT_EDITION } from './constants';
 
 import ParserCtx from './parser';
 import { isReservedES6Keyword } from './utils/javascript';
 
 export const EXPRESSION_SYMBOL_START = '{';
 export const EXPRESSION_SYMBOL_END = '}';
 
-const VALID_EXPRESSION_RE = /^{.+}$/;
 const POTENTIAL_EXPRESSION_RE = /^.?{.+}.*$/;
 const WHITESPACES_RE = /\s/;
 
 export function isExpression(source: string): boolean {
-    return !!source.match(VALID_EXPRESSION_RE);
+    return source[0] === '{' && source.slice(-1) === '}';
```

Comment:
I don't believe this should be an observable/breaking change. The only difference between these two conditions is whether `source` contains a newline. However, unquoted attribute values cannot contain a new line - so the two are functionally equivalent with the flag disabled.

With the flag enabled, they are not functionally equivalent - the new condition will permit a newline in the attribute value, which is the desired behavior.

Commenting for visibility, in case there is something I'm not considering.